src/main/java/seedu/command/AddExpenseCommand.java

```diff
@@ -0,0 +1,107 @@
+package seedu.command;
+
+import seedu.category.Category;
+import seedu.transaction.Expense;
+import seedu.transaction.Transaction;
+import seedu.transaction.TransactionList;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class AddExpenseCommand extends AddTransactionCommand {
+    public static final String COMMAND_WORD = "add-expense";
+    public static final String COMMAND_GUIDE = "add-expense [DESCRIPTION] [a/ AMOUNT] [d/ DATE] [c/ hCATEGORY]";
+    public static final String[] COMMAND_MANDATORY_KEYWORDS = {"a/"};
+    public static final String[] COMMAND_EXTRA_KEYWORDS = {"d/", "c/"};
+
+    public static final String ERROR_MESSAGE = "Error creating Income!";
+
+    public AddExpenseCommand(TransactionList transactions) {
+        super(transactions);
+    }
+
+    @Override
+    public List<String> execute() {
```

Comment:
This function seems to have multiple functionalities embedded. You can consider refactoring this.

Reply:
Created a new issue. Thank you!